NPE in OpenApiAnnotationScanner.createResponseFromJaxRsMethod

Using a very simple JAX-RS resource that returns a JSON-P object, SmallRye OpenAPI throws a NPE:

```java
@Path("/")
public class HelloResource {
    @GET
    public JsonObject hello() {
        return Json.createObjectBuilder()
                .add("content", "foobar")
                .build();
    }
}
```

```
2020-01-30 19:06:44,774 ERROR [stderr] (main) Caused by: java.lang.NullPointerException
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.createResponseFromJaxRsMethod(OpenApiAnnotationScanner.java:934)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.processJaxRsMethod(OpenApiAnnotationScanner.java:688)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.lambda$processJaxRsResourceClass$0(OpenApiAnnotationScanner.java:422)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1556)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
2020-01-30 19:06:44,775 ERROR [stderr] (main)   at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.processJaxRsResourceClass(OpenApiAnnotationScanner.java:420)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.scan(OpenApiAnnotationScanner.java:221)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at io.smallrye.openapi.runtime.OpenApiProcessor.modelFromAnnotations(OpenApiProcessor.java:72)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at org.wildfly.swarm.microprofile.openapi.runtime.OpenApiDeploymentProcessor.process(OpenApiDeploymentProcessor.java:101)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at org.wildfly.swarm.microprofile.openapi.runtime.OpenApiDeploymentProcessor$Proxy$_$$_WeldClientProxy.process(Unknown Source)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   at org.wildfly.swarm.container.runtime.RuntimeDeployer.deploy(RuntimeDeployer.java:233)
2020-01-30 19:06:44,776 ERROR [stderr] (main)   ... 12 more
```

As you can see in the stacktrace, this is using Thorntail. Specifically, Thorntail 2.6.0.Final, which uses SmallRye OpenAPI 1.1.20. I've tried to rebuild Thorntail with SmallRye OpenAPI 1.1.10, that works well. I've also tried to build SmallRye OpenAPI `master` branch and integrate that into Thorntail, same error.

I'm attaching a simple Thorntail-based reproducer.

[reproducer.zip](https://github.com/smallrye/smallrye-open-api/files/4135632/reproducer.zip)

--- a/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java
+++ b/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/OpenApiAnnotationScanner.java
@@ -732,7 +732,6 @@ public class OpenApiAnnotationScanner {
 
         // Process @APIResponse annotations
         /////////////////////////////////////////
-        APIResponses responses = null;
         List<AnnotationInstance> apiResponseAnnotations = JandexUtil.getRepeatableAnnotation(method,
                 OpenApiConstants.DOTNAME_API_RESPONSE, OpenApiConstants.DOTNAME_API_RESPONSES);
         for (AnnotationInstance annotation : apiResponseAnnotations) {
@@ -1036,7 +1035,7 @@ public class OpenApiAnnotationScanner {
                     produces = OpenApiConstants.DEFAULT_MEDIA_TYPES.get();
                 }
 
-                if (schema.getNullable() == null && TypeUtil.isOptional(returnType)) {
+                if (schema != null && schema.getNullable() == null && TypeUtil.isOptional(returnType)) {
                     schema.setNullable(Boolean.TRUE);
                 }
 

--- a/implementation/src/test/java/io/smallrye/openapi/runtime/scanner/ResourceParameterTests.java
+++ b/implementation/src/test/java/io/smallrye/openapi/runtime/scanner/ResourceParameterTests.java
@@ -322,4 +322,35 @@ public class ResourceParameterTests extends OpenApiDataObjectScannerTestBase {
             return null;
         }
     }
+
+    /*************************************************************************/
+
+    /*
+     * Test case derived from original example in SmallRye OpenAPI issue #248.
+     *
+     * https://github.com/smallrye/smallrye-open-api/issues/248
+     *
+     */
+    @Test
+    public void testResponseTypeUnindexed() throws IOException, JSONException {
+        // Index is intentionally missing ResponseTypeUnindexedTestResource$ThirdPartyType
+        Index index = indexOf(ResponseTypeUnindexedTestResource.class);
+        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(emptyConfig(), index);
+        OpenAPI result = scanner.scan();
+        printToConsole(result);
+        assertJsonEquals("responses.unknown-type.empty-schema.json", result);
+    }
+
+    @Path("/unindexed")
+    static class ResponseTypeUnindexedTestResource {
+        // This type will not be in the Jandex index, nor does it implement Map or List.
+        static class ThirdPartyType {
+        }
+
+        @GET
+        @Produces(MediaType.TEXT_PLAIN)
+        public ThirdPartyType hello() {
+            return null;
+        }
+    }
 }

--- a/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.unknown-type.empty-schema.json
+++ b/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/responses.unknown-type.empty-schema.json
@@ -1,0 +1,19 @@
+{
+  "openapi": "3.0.1",
+  "paths": {
+    "/unindexed": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}

